libbpf-rs/src/object.rs

```diff
@@ -225,6 +270,14 @@ impl OpenObject {
         self.map(name).unwrap().unwrap()
     }
 
+    /// Get an iterator over references to all `OpenMap`s.
+    /// Note that this will include automatically generated .data, .rodata, .bss, and
+    /// .kconfig maps.
+    pub fn maps_iter(&mut self) -> impl Iterator<Item = &mut OpenMap> {
```

Comment:
I'm wondering how mutable and read-only iterators are generally implemented? E.g., if I just want to print out all map and prog names and I don't have mutable reference to OpenObject, how can I do that?

Reply:
That would be with an iterator that returns an immutable reference. We discussed this above somewhere. Right now I'm thinking it makes more sense to include such functionality in a second PR.

Reply:
Signature would be like this: `pub fn maps_iter(&self) -> impl Iterator<Item = &OpenMap>` and then the body would just return `self.maps.values()`.

Reply:
My plan is to eventually propose changing all existing map and prog getters to have a `_mut` suffix and writing an immutable version for each. This goes a bit beyond the scope of this PR, which is why I wanted to do it in a follow-up.

Reply:
I was more concerned with naming, because I think Rust doesn't allow method overrides. But I guess we are still at the point where it's not too bad to change APIs and rename them, so ok.